Pasteboard Viewer/ContentsScreen.swift

```diff
@@ -1,12 +1,66 @@
 import SwiftUI
 import UniformTypeIdentifiers
+import HexFiend
 
 struct ContentsScreen: View {
 	@EnvironmentObject private var pasteboardObservable: XPasteboard.Observable
+	@State private var viewAsText = true
+	@State private var selectedByteRanges = Set<Range<UInt64>>()
 
 	let type: Pasteboard.Type_
 
 	var body: some View {
+		Group {
+			if viewAsText {
+				textBody
+			} else {
+				hexBody
+			}
+		}
+		.fillFrame()
+		.navigationTitle(type.decodedDynamicTitleIfAvailable ?? type.title)
+		.toolbar {
+			formatPicker
+			moreButton
+		}
+	}
+
+	private var hexBody: some View {
+		// this uses a .number formatter instead of .byteCount,
+		// because a byte count format doesn't have a way to exclude the grouping character
+		let count = (type.data()?.count ?? 0)
+		let length = count.formatted(.number.grouping(.never)) + " " + (count == 1 ? "byte" : "bytes")
+		let extraInfo: String
+		if selectedByteRanges.isEmpty {
+			// no selection
+			extraInfo = length
+		} else if let range = selectedByteRanges.first, selectedByteRanges.count == 1 {
+			// single or empty selection
+			let offset = range.lowerBound.formatted(.number.grouping(.never))
+			if range.lowerBound == range.upperBound {
+				// no selection
+				extraInfo = length
+			} else if range.upperBound - range.lowerBound == 1 {
+				// single byte
+				extraInfo = "Byte \(offset) selected out of \(length)"
+			} else {
+				// multiple bytes
+				let count = (range.upperBound - range.lowerBound).formatted(.number.grouping(.never))
+				extraInfo = "\(count) bytes selected at offset \(offset) out of \(length)"
+			}
+		} else {
+			// multiple selection
+			let totalCount = selectedByteRanges.reduce(into: UInt64.zero, { $0 += ($1.upperBound - $1.lowerBound) })
+				.formatted(.number.grouping(.never))
+
+			extraInfo = "\(totalCount) selected at multiple offsets out of \(length)"
+		}
```

Comment:
I would move this to a `hexBodyExtraInfo` getter to improve readability as then each branch could use `return`. And it could also use `guard`.